quantum/rgb_matrix/animations/pixel_flow_anim.h

```diff
@@ -0,0 +1,51 @@
+// Copyright 2022 @filterpaper
+// SPDX-License-Identifier: GPL-2.0+
+
+#ifdef ENABLE_RGB_MATRIX_PIXEL_FLOW
+RGB_MATRIX_EFFECT(PIXEL_FLOW)
+#    ifdef RGB_MATRIX_CUSTOM_EFFECT_IMPLS
+
+static bool PIXEL_FLOW(effect_params_t* params) {
+    // LED state array
+    static RGB led[DRIVER_LED_TOTAL];
```

Comment:
The consumption of RAM by this array may be significant (especially if using AVR chips). I wonder whether we should somehow optimize this by making the buffers used by different effects overlap (because at the moment only one effect at a time can run). Note that the `PIXEL_FRACTAL` effect also uses a RAM buffer with a comparable size (that one uses the key matrix size though).

Reply:
Memory impact was reviewed with `avr-size` while coding that array table. This PCB uses 58 LEDs and the effect footprint was 178 bytes of data memory:
Without PIXEL_FLOW:
```
avr-size -C --mcu=atmega32u4 .build/boardsource_technik_o_default.elf
AVR Memory Usage
----------------
Device: atmega32u4

Program:   23712 bytes (72.4% Full)
(.text + .data + .bootloader)

Data:        845 bytes (33.0% Full)
(.data + .bss + .noinit)
```
WIth PIXEL_FLOW:
```
avr-size -C --mcu=atmega32u4 .build/boardsource_technik_o_default.elf
AVR Memory Usage
----------------
Device: atmega32u4

Program:   24130 bytes (73.6% Full)
(.text + .data + .bootloader)

Data:       1023 bytes (40.0% Full)
(.data + .bss + .noinit)
```

Is this a good benchmark for evaluating RAM impact? I can try to rework this to reduce the array at the expense of more code.

Reply:
Let's revisit a "scratch" buffer for all the effects in a followup PR.